src/views/Routes.vue

```diff
@@ -301,6 +308,11 @@ export default class Routes extends Mixins(Perf, UserLoader) {
         );
         logError('set_climb_state_failed', err);
       });
+
+    this.pendingWrites++;
+    this.firestore.waitForPendingWrites().then(() => {
+      this.pendingWrites--;
```

Comment:
Should we check that this doesn't go below 0 due to some error?

Reply:
i don't think that that's possible, since a promise can only be resolved a single time.